[otbn,dv] V2S Testplan Update

ctopal opened the pull request:
Commit includes mostly description updates on the existing list. It also includes some additions to the test list for LOCAL_ESC tests and a change in KEY.SIDELOAD to point to otbn_single since in that we are guaranteed to program OTBN.

Signed-off-by: Canberk Topal <ctopal@lowrisc.org>

weicaiyang commented on hw/ip/otbn/data/otbn_sec_cm_testplan.hjson:
```diff
@@ -167,86 +224,126 @@
-            However, from defined CSRs and memory returned data, there is no way to read
-            scramble key by SW.
+      desc: '''Verify the countermeasure(s) KEY.SIDELOAD.
+            Invalid sideload keys are allowed in the sideload key sequence fifty percent of the
+            time by default. That verifies the sceneario when in the case of an invalid sideload
+            key, OTBN would generate a KEY_INVALID recoverable software error.
+            This happens test agnostic so otbn_single is mapped to represent an OTBN run in general.
```
The spec mentions `Keys can be sideloaded without exposing them to the CPU` for this countermeasure.

IMO, DV can't really verify that. @GregAC can also confirm. Perhaps we can mention it here. I'm fine to link the test that uses the sideload key.

weicaiyang reviewed (approved):
LGTM with a suggestion

andreaskurth commented on hw/ip/otbn/data/otbn_sec_cm_testplan.hjson:
```diff
@@ -25,119 +25,172 @@
       name: sec_cm_start_stop_ctrl_fsm_global_esc
-      desc: "Verify the countermeasure(s) START_STOP_CTRL.FSM.GLOBAL_ESC."
+      desc: '''Verify the countermeasure(s) START_STOP_CTRL.FSM.GLOBAL_ESC.
+
```
```suggestion
            Run an OTBN program, drive lc_escalate_en_i port randomly to
            see global escalation locking up the start-stop control FSM in OTBN.
```

andreaskurth reviewed (approved):
Generally LGTM, thanks @ctopal!

I think `sec_cm_start_stop_ctrl_fsm_global_esc` is still missing a description and have suggested one

GregAC commented on hw/ip/otbn/data/otbn_sec_cm_testplan.hjson:
```diff
@@ -25,119 +25,172 @@
       name: sec_cm_data_mem_integrity
-      desc: "Verify the countermeasure(s) DATA.MEM.INTEGRITY."
+      desc: '''Verify the countermeasure(s) DATA.MEM.INTEGRITY.
+            Run an OTBN program multiple times and corrupt DMEM registers
```
DMEM is not a register so just 'corrupt DMEM' or 'corrupt the DMEM'?

GregAC commented on hw/ip/otbn/data/otbn_sec_cm_testplan.hjson:
```diff
@@ -25,119 +25,172 @@
       name: sec_cm_instruction_mem_integrity
-      desc: "Verify the countermeasure(s) INSTRUCTION.MEM.INTEGRITY."
+      desc: '''Verify the countermeasure(s) INSTRUCTION.MEM.INTEGRITY.
+            Run an OTBN program multiple times and corrupt IMEM registers
```
As above, IMEM is not registers.

GregAC commented on hw/ip/otbn/data/otbn_sec_cm_testplan.hjson:
```diff
@@ -25,119 +25,172 @@
     }
     {
       name: sec_cm_start_stop_ctrl_fsm_local_esc
       desc: "Verify the countermeasure(s) START_STOP_CTRL.FSM.LOCAL_ESC."
```
Don't we want a similar expansion of the description as we had for SCRAMBLE_CTRL.FSM.LOCAL_ESC?

GregAC reviewed (approved):
Couple of tweaks to do but otherwise LGTM

ctopal commented:
Thanks everyone for the feedback, I implemented the suggestions and added an explanation about the actual countermeasure itself being not verifiable with the simulation in the sideload keys case. I'm planning to merge this after CI passes so please let me know if that's not okay.
